docker-compose.ytt.yaml

```diff
@@ -135,6 +135,29 @@ services:
     volumes:
       - db-data:/var/lib/postgresql/data
       - ./postgres/init:/docker-entrypoint-initdb.d/
+  
+  virtuoso:
+    image: tenforce/virtuoso:latest
+    environment: 
+      SPARQL_UPDATE: "true"
+      VIRT_$Database_DatabaseFile: /data/virtuoso_db/virtuoso.db
+      VIRT_$Database_ErrorLogFile: /data/virtuoso.log
+      VIRT_$Database_LockFile: /data/virtuoso.lck
+      VIRT_$Database_ErrorLogLevel: 7
+      VIRT_$Database_FileExtend: 200
+      VIRT_$Database_MaxCheckpointRemap: 2000
+      VIRT_$Database_Striping: 0
+      VIRT_$Database_TempStorage: TempDatabase
+      VIRT_$Parameters_O_DIRECT: 1
+      VIRT_$Parameters_ServerThreads: 100
+      VIRT_$HTTPServer_ServerThreads: 15
+      VIRT_$SPARQL_MaxQueryCostEstimationTime: 600
+      VIRT_$SPARQL_MaxQueryExecutionTime: 180
+
+    volumes:
+      - ./virtuoso_db:/data/virtuoso_db
+    ports:
```

Comment:
Does it need to be exposed to the host? I suppose the skills, datastores and other services use the docker network to communicate with it.

Reply:
Yes. I also need to query it from external networks to do some research work.